src/evaluators/DoWhileStatement.js

```diff
@@ -79,8 +79,8 @@ export default function(
   let result = realm.evaluateForFixpointEffects(iteration);
   if (result !== undefined) {
     let [outsideEffects, insideEffects, cond] = result;
-    let [rval] = outsideEffects.data;
-    let [, bodyGenerator] = insideEffects.data;
+    let { result: rval } = outsideEffects;
```

Comment:
I'm all for consistency, but I do have to wonder if it would be more readable to just use the property directly, i.e.
let rval = outsideEffects.result;